potodo/potodo.py

```diff
@@ -4,224 +4,189 @@
+        print(f"\n\n# {directory_name} ({statistics.mean(folder_stats):.2f}% done)\n")
+        print("\n".join(buffer))
+
+
+def buffer_add(
+    buffer: list,
+    folder_stats: list,
+    printed_list: list,
+    po_file: PoFile,
+    issue_reservations: dict,
     above: int,
     below: int,
-    matching_files: bool,
-    fuzzy: bool,
-    offline: bool,
-    hide_reserved: bool,
-):
+) -> Tuple[list, list, list]:
+    """
+    Will add to the buffer the information to print about the file is the file isn't translated entirely or above or below requested values
+    """
+    if po_file.percent_translated == 100:
+        # If the file is completely translated
+
+        # Add the percentage of the file to the stats of the folder
+        folder_stats.append(po_file.percent_translated)
+        # Indicate not to print that file
+        printed_list.append(False)
+        # End the function call without adding anything to the buffer
+        return buffer, folder_stats, printed_list
+
+    if po_file.percent_translated < above:
+        # If the file's percent translated is below what is requested
+
+        # Add the percentage of the file to the stats of the folder
+        folder_stats.append(po_file.percent_translated)
+        # Indicate not to print that file
+        printed_list.append(False)
+        # End the function call without adding anything to the buffer
+        return buffer, folder_stats, printed_list
+
+    if po_file.percent_translated > below:
+        # If the file's percent translated is above what is requested
+
+        # Add the percentage of the file to the stats of the folder
+        folder_stats.append(po_file.percent_translated)
+        # Indicate not to print that file
+        printed_list.append(False)
+        # End the function call without adding anything to the buffer
+        return buffer, folder_stats, printed_list
+
+    buffer.append(
+        # The filename
+        f"- {po_file.filename:<30} "
+        # The number of entries translated / the file size
+        + f"{po_file.translated_nb:3d} / {po_file.po_file_size:3d} "
+        # The percent of the file translated
+        + f"({po_file.percent_translated:5.1f}% translated)"
+        # The fuzzies in the file IF fuzzies exists in the file
+        + (f", {po_file.fuzzy_nb} fuzzy" if po_file.fuzzy_entries else "")
+        # The `reserved by` if the file is reserved unless if the offline/hide_reservation are enabled
+        + (
+            f", réservé par {issue_reservations[po_file.filename_dir.lower()]}"
+            if po_file.filename_dir.lower() in issue_reservations
+            else ""
+        )
+    )
+    # Add the percent translated to the folder statistics
+    folder_stats.append(po_file.percent_translated)
+    # Indicate to print the file
+    printed_list.append(True)
+
+    # Return the updated vars with the new file
+    return buffer, folder_stats, printed_list
```

Comment:
If you mutate the buffer, please don't return it.

A function returning something is telling "I created a new one", a function returning None is telling "I mutated your arguments", see sorted (returning a new list, not touching the given one) vs list.sort (returning None, sorting inplace).

Reply:
So should I just return None and if I mutate the var inside my function, once I exit the function the var will still be altered ? 🤔

Reply:
ping @JulienPalard

Reply:
Yes, lists, like literally anything, is passed by reference in Python, so:
```python
def a(l): l.append(42)
l = []
a(l)
print(l)
```
gives `[42]`

Reply:
I hate C

Reply:
so, remind me the difference between `const char *c`, `char const *c` and `char * const c`?

Reply:
I don't have a single freaking clue man